test-data/unit/check-inference.test

```diff
@@ -2303,3 +2303,97 @@ class C:
     def f(self, x) -> None:
         # TODO: It would be better for the type to be Any here
         self.a.y  # E: "None" has no attribute "y"
+
+-- Special case for assignment to '_'
+-- ----------------------------------
+
+[case testUnusedTargetSingle]
+def foo() -> None:
```

Comment:
Why do you have only tests at the function scope? Is it something intentional?

Reply:
Oops, I thought it would all be treated the same, but it isn't. Will fix. (Since I've seen a fair number of global uses of this idiom.)